### PR TITLE
Fix united sweep

### LINKED ISSUE #617: sweeps fail with quantity substitutions
MWE:

```python
import numpy as np
from gpkit import Variable, Model, units

A = Variable("A", "USD")
h = Variable("h", "USD/count")
Q = Variable("Q", "count")
Y = Variable("Y", "USD")

m = Model(Y, [Y >= h*Q + A/Q])
m.substitutions.update({A: 500*units("USD"),
                        h: 35*units("USD"),
                        Q: ("sweep", [50, 100, 500])})
sol = m.solve()
```
results in `ValueError: setting an array element with a sequence.`. 

- [x] add unit test for this issue

### PR DESCRIPTION
**Pull request by bqpd**
Closes #617

### CODE REVIEW

**Comment by bqpd** [comment]
@whoburg, ready for review

**Comment by whoburg** [comment]
retest this please

File: gpkit/small_scripts.py [hunk 1]
```diff
@@ -56,6 +56,8 @@ def unitstr(units, into="%s", options="~", dimless='-'):
     if units and not isinstance(units, Strings):
         try:
             rawstr = ("{:%s}" % options).format(units)
+            if str(units.units) == "count":
+                rawstr = "1.0 count"
```
Comment (whoburg) [inline gpkit/small_scripts.py]: I don't understand why this special case is needed. Using the version on master (without the change above), unitstr seems to work fine:

```python
In [1]: from gpkit import units

In [2]: from gpkit.small_scripts import unitstr

In [3]: from gpkit import Variable

In [4]: x = Variable("x", "ft")

In [5]: x.units
Out[5]: <Quantity(1, 'foot')>

In [6]: unitstr(x)
Out[6]: 'ft'

In [7]: x = Variable("x", "count")

In [8]: unitstr(x)
Out[8]: 'count'
```

File: gpkit/small_classes.py [hunk 2]
```diff
@@ -139,12 +139,15 @@ def _enray_and_unit_dict(d_in, d_out, unitless_keys=(), united=False):
             d_out[k] = _enray_and_unit_dict(v, v.__class__(),
                                             unitless_keys, united)
         else:
+            if hasattr(v[0], "units"):
```
Comment (whoburg) [inline gpkit/small_classes.py]: inline comment would be helpful here

**Comment by whoburg** [comment]
reviewed -- currently failing unit and dependency tests

File: gpkit/small_scripts.py (hunk 1 above)
Comment (bqpd) [inline gpkit/small_scripts.py]: Hmm, it wasn't working in printing solution arrays without that; they kept printing just as dimensionless.

**Comment by bqpd** [comment]
retest this please

**Comment by bqpd** [comment]
@whoburg: ready for re-review

**Comment by bqpd** [comment]
merge if passes tests

**Pull request merged**